rewrite-core/src/main/java/org/openrewrite/Recipe.java

```diff
@@ -300,7 +300,10 @@ public Validated<Object> validate() {
         List<Field> requiredFields = NullUtils.findNonNullFields(this.getClass());
         for (Field field : requiredFields) {
             try {
-                validated = validated.and(Validated.required(field.getName(), field.get(this)));
+                validated = validated.and(
+                        field.getType().equals(String.class)
+                                ? Validated.notBlank(field.getName(), (String) field.get(this))
+                                : Validated.required(field.getName(), field.get(this)));
```

Comment:
Hmm I'm not sure about imposing a non-blankness default on all recipes. I don't mind the rest of this, but making allowing the empty string opt-in would be surprising to recipe authors.

Reply:
That's fair. Any opinion on the alternatives discussed in #3508 ?

Reply:
pushed a change